staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -78,6 +78,9 @@ type Config struct {
 	// this to confirm the proxy's identity
 	ProxyClientCert []byte
 	ProxyClientKey  []byte
+
+	// Indicates if the Aggregator should send to the cluster IP (false) or route to the endpoints IP (true)
+	EnableAggregatorRouting bool
```

Comment:
Added two levels of interface to handle the explicit hostname when routing is disabled.